recipes/soci/all/conanfile.py

```diff
@@ -0,0 +1,187 @@
+
+        self._cmake.definitions["WITH_BOOST"] = self.options.with_boost
+        self._cmake.definitions["SOCI_EMPTY"] = self.options.with_backend_empty
+        self._cmake.definitions["WITH_SQLITE3"] = self.options.with_backend_sqlite3
+        self._cmake.definitions["WITH_MYSQL"] = self.options.with_backend_mysql
+        self._cmake.definitions["WITH_ODBC"] = self.options.with_backend_odbc
+        self._cmake.definitions["WITH_POSTGRESQL"] = self.options.with_backend_postgresql
+
+        self._cmake.definitions["WITH_DB2"] = self.options.with_backend_db2
+        self._cmake.definitions["WITH_FIREBIRD"] = self.options.with_backend_firebird
+        self._cmake.definitions["WITH_ORACLE"] = self.options.with_backend_oracle
+
+        self._cmake.configure()
+        return self._cmake
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+        if self.options.with_backend_sqlite3:
+            tools.replace_in_file("FindSQLite3.cmake", "SQLite3_FOUND", "SQLITE3_FOUND")
+            tools.replace_in_file("FindSQLite3.cmake", "SQLite3_INCLUDE_DIRS", "SQLITE3_INCLUDE_DIRS")
+            tools.replace_in_file("FindSQLite3.cmake", "SQLite3_LIBRARIES", "SQLITE3_LIBRARIES")
+        if self.options.with_backend_mysql:
+            os.rename("Findlibmysqlclient.cmake", "FindMySQL.cmake")
+            tools.replace_in_file("FindMySQL.cmake", "libmysqlclient_FOUND", "MYSQL_FOUND")
+            tools.replace_in_file("FindMySQL.cmake", "libmysqlclient_INCLUDE_DIRS", "MYSQL_INCLUDE_DIRS")
+            tools.replace_in_file("FindMySQL.cmake", "libmysqlclient_LIBRARIES", "MYSQL_LIBRARIES")
+        if self.options.with_backend_postgresql:
+            tools.replace_in_file("FindPostgreSQL.cmake", "PostgreSQL_FOUND", "POSTGRESQL_FOUND")
+            tools.replace_in_file("FindPostgreSQL.cmake", "PostgreSQL_INCLUDE_DIRS", "POSTGRESQL_INCLUDE_DIRS")
+            tools.replace_in_file("FindPostgreSQL.cmake", "PostgreSQL_LIBRARIES", "POSTGRESQL_LIBRARIES")
+
+    def build(self):
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE_1_0.txt", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        tools.rmdir(os.path.join(self.package_folder, "cmake"))
+
+    def _construct_library_name(self, name):
+        if self.settings.os == "Windows":
+            abi_version = tools.Version(self.version)
+            name = "lib{name}_{major}_{minor}".format(
+                name = name,
+                major = abi_version.major,
+                minor = abi_version.minor
+            )
+        return name
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "SOCI"
+        self.cpp_info.names["cmake_find_package_multi"] = "SOCI"
```

Comment:
The name for `cmake_find_package` should be `Soci`. At least this is how I find it when looking for CMake modules: `FindSoci.cmake`

Reply:
Done